### DISCUSSION TITLE
这周你做了什么？2022-04-08

### DISCUSSION THREAD

**Discussion by jiacai2050** [EmacsTalk/emacstalk.github.io#12] [Show and tell]
- 可以随便讨论本周你打算做什么与 Emacs 相关的事情，甚至可以寻求帮助或反馈。
- 请记住，什么都不做也没关系！
- Happy Hacking Emacs！

**Comment by jiacai2050** [comment]
- 清明节期间和懒猫录制节目，本想留着五一再发布，但实在是没忍住，于是花了一天剪了出来，目前为止最长的一期，差不多两个小时
  -  https://emacstalk.github.io/podcast/012/
- 重新写了个 project eshell，方便在大屏上左右分屏用
```elisp
(defun my/project-eshell ()
  (interactive)
  (let* ((buf-name (project-prefixed-buffer-name "eshell")))
    (if-let ((win (get-buffer-window buf-name)))
        (delete-window win)
      (when (= (length (window-list)) 1)
        (split-window-horizontally))
      (other-window 1)
      (project-eshell))))

```

**Comment by VSteveHL** [comment]
1. 试试看能不能入坑 [evil-mode](https://github.com/emacs-evil/evil)
2. 想着折腾下 modeline 和 theme，同时逛逛社区看看大家对美化 Emacs 的看法

**Reply by jiacai2050** [reply]
熟悉 vi 的话，evil 没什么难度，就是有些按键需要调整下。可参考
- https://github.com/noctuid/evil-guide#keymap-precedence

**Reply by VSteveHL** [reply]
主要是对 vim 也不熟悉... 也就局限于 hjkl 和 :wq 这种最基础的指令上了。当时为了 org-mode 直接入坑了 Emacs， 这么久用下来也习惯 Emacs 这套按键了（把大写锁定键和 Ctrl 交换了，感觉还行）。现在想着还是找点原生的 vim 教程，看看能不能把 vim 用起来，直接上 evil-mode 如果搞出来什么问题感觉我这种初学者也不好定位到底是自己 vim 模式不熟悉还是 evil-mode 没配置好。

我是在 [一年成为Emacs高手](https://github.com/redguardtoo/mastering-emacs-in-one-year-guide/blob/master/guide-zh.org) 这篇文章里看到的 evil-mode，作者挺推崇这个的，但是作者也提到他是精通 Vi 之后才转到 Emacs 里的，我这种小白估计一时半会也体会不到 vim 模式的魅力，真用不来就算了，算了...

**Reply by jiacai2050** [reply]
之前不熟悉 vim 的话，确实可以跳过 evil 了，收益不大。

**Reply by AmaiKinono** [reply]
我的情况也是先接触 Emacs，试着入坑 evil，后来发现自己基本上只是用 h/j/k/l、C-d/u/f/b 这些命令，其实就是想要个类似 less 的浏览模式。之后发现 Emacs 自带的 view-mode 就很不错。

**Reply by VSteveHL** [reply]
我是觉得 Vim 里用 "daw" 来删除一个单词挺舒服的（虽然用 M-b M-d 两个键也能做到删除一个单词），不知道 Emacs 里面有没有这种能对一个指定的范围进行操作的，比如写 Lisp 的时候用一个快捷键就可以删除掉一对括号及其里面的内容这种。

查了查其实就是想说 Emacs 里面有没有 Text Object 这种东西，不知道除了 evil 之外还有没有类似的解决方案。（不过现在用上了 ParEdit ，自己胡乱写点 Lisp 玩足够用了，也没有特别强的需求）

**Reply by jiacai2050** [reply]
我一般都是用 M-f M-DEL 来删除，搜了下，想达到 Vim 里面的 daw 也很好做。
- https://stackoverflow.com/a/33450643/2163429

```el
(defun my/kill-thing-at-point (thing)
  "Kill the `thing-at-point' for the specified kind of THING."
  (let ((bounds (bounds-of-thing-at-point thing)))
    (if bounds
        (kill-region (car bounds) (cdr bounds))
      (error "No %s at point" thing))))

(defun my/kill-word-at-point ()
  "Kill the word at point."
  (interactive)
  (my/kill-thing-at-point 'word))
```

**Reply by VSteveHL** [reply]
又学到一个新函数 🤣

**Comment by thomjiji** [comment]
想在 Org mode 里更好的输入中文，首先一个问题需要解决：中英相邻的时候我习惯手动键入空格，这样会导致 Emacs 在空格的地方强行换行。看到 Emacs China 论坛上有[关于此](https://emacs-china.org/t/topic/2616/18)的很多讨论和解决办法，还在学习中。不知道有没更简单易上手的办法。

**Reply by AmaiKinono** [reply]
试一下 28 新加的 `word-wrap-by-category`？

**Comment by KngZhi** [comment]
之前用 Emacs 写 Org File 卡到爆炸，后来查了一下 Profile 发现 Company-mode 占用了过多的内存和 CPU。关了之后，芜湖，起飞 🛫️

**Reply by thomjiji** [reply]
不用 Company 的话有什么别的替代方案嚜

**Reply by KngZhi** [reply]
可能无法解答你这个问题。

因为我写 Org 不太需要补全。如果写代码的话就用 nvim, emacs 中 lsp 的补全也非常卡。

之前在 Emacs-China 中看到，懒猫说之后有空的话可能会改进 lsp 卡顿的问题，可以期待一下吧。

**Reply by jiacai2050** [reply]
用 eglot 就好了。
- https://emacstalk.github.io/post/016/

**Comment by yibie** [comment]
本周已经执行学习李杀 Elisp 教程的第一章，目前已经学完数据类型部分，已经将学习帖子发布在 Emacs-China 社区：https://emacs-china.org/t/elisp/18752/9

接下来今天打算学完 Elisp Basic。
